Assets/Scripts/Models/HealthSystem.cs

```diff
@@ -0,0 +1,117 @@
+﻿using System;
+
+[Serializable]
+public class HealthSystem
+{
+    //all of this data should probably be generated at runtime or loaded from a file
+    public float Value
+    {
+        get
+        {
+            return _currentHealth;
+        }
+
+        private set
+        {
+            if (Invincible == true) return;
+            //We have been Revived!
+            if (_currentHealth <= 0 && value > _currentHealth)
+            {
+                _currentHealth = value;
+
+                if (_OnRevive != null)
+                    _OnRevive();
+                return;
+            }
+            //we have been hit but we are not dead!
+            if (value < _currentHealth && value > 0)
+            {
+                _currentHealth = value;
+                if (_OnHit != null)
+                    _OnHit();
+            }
+            //we have been healed!
+            if (value > _currentHealth)
+            {
+                if (value > MaxHealth)
+                    value = MaxHealth;
+                _currentHealth = value;
+                if (_OnHeal != null)
+                    _OnHeal();
+
+            }
+            //We have died and this means we have been hit!
+            if (value <= 0 && IsAlive)
+            {
+                _currentHealth = 0;
+                if (_OnHit != null)
+                    _OnHit();
+                if (_OnDestruction != null)
+                    _OnDestruction();
+            }
+        }
+    }
+    public float MaxHealth { get; set; }
+    public bool Invincible { get; set; }
+    public bool Healable { get; set; }
+    public bool IsAlive { get { return Value > 0; } }
+
+    private float _currentHealth = 100.0f;
+    private Action _OnHit;
```

Comment:
I might have just learned something ^^

Anyways if you declare them as events:
```csharp
public event Action OnHit;
```

You don't need to manually protect them:
http://stackoverflow.com/questions/29155/what-are-the-differences-between-delegates-and-events

Events are a compiler feature to simplify exactly this use case.